### PR TITLE
TJT-1921 Address Opt In

### PR DESCRIPTION
**Pull request by msaadq**
# Why?

https://finn-jira.atlassian.net/browse/TJT-1921

# What?

* [Improve the alignment of Checkbox by reducing the extra paddings from the base view](https://github.com/finn-no/FinniversKit/commit/c5fde0dba06f56ebf3a1b7c8cd3c2abe3fcf1282)
* [Modify the BlockUserView which depends on Checkbox and Update Snapshots](https://github.com/finn-no/FinniversKit/commit/355aa20f6c87d99b3a92dd5b4fdff5b878cc2477)

# Version Change

minor

### CODE REVIEW

File: FinniversKit/Sources/Components/SelectionBoxes/Selectionbox.swift [hunk 1]
```diff
@@ -149,18 +155,24 @@ extension Selectionbox {
...
-        addSubview(stack)
+        
+        addSubview(containerStack)
+        
+        containerStack.addArrangedSubviews([
+            titleLabel,
+            stack
+        ])
+        
+        if titleLabel.text == nil {
```
Comment (BjornRuud) [inline FinniversKit/Sources/Components/SelectionBoxes/Selectionbox.swift:166]: If you set the label to hidden the stack view will hide it and adjust spacing automatically.

File: FinniversKit/Sources/Components/SelectionBoxes/Selectionbox.swift [hunk 2]
```diff
@@ -78,7 +78,10 @@ public class Selectionbox: UIView {
 
     public var title: String? {
         get { return titleLabel.text }
-        set { titleLabel.text = newValue }
+        set {
+            titleLabel.text = newValue
+            setupSubviews()
```
Comment (BjornRuud) [inline FinniversKit/Sources/Components/SelectionBoxes/Selectionbox.swift:83]: Not necessary to do a new setup each time the title is set.

**Review by BjornRuud — CHANGES_REQUESTED** [review]
Some comments. Since this is FinniversKit the PR should also be reviewed by someone outside TJT.

File: FinniversKit/Sources/Components/SelectionBoxes/Selectionbox.swift (hunk 2 above)
Comment (msaadq) [reply FinniversKit/Sources/Components/SelectionBoxes/Selectionbox.swift:83]: We need to hide the title if the label text is set to nil

Comment (BjornRuud) [reply FinniversKit/Sources/Components/SelectionBoxes/Selectionbox.swift:83]: Yes but the label can just toggle isHidden based on presence of the text.

**Review by BjornRuud — APPROVED** [review]
👍

**Pull request merged**